Frameworks/Foundation/NSMutableArray.mm

```diff
@@ -157,46 +158,59 @@ - (void)exchangeObjectAtIndex:(NSUInteger)atIndex withObjectAtIndex:(NSUInteger)
 /**
  @Status Interoperable
 */
-- (void)removeObject:(NSObject*)objAddr {
-    if (objAddr == nil) {
-        TraceVerbose(TAG, L"objAddr = nil!");
-    }
-
-    int idx = [self indexOfObject:objAddr];
-    if (idx != NSNotFound) {
-        [self removeObjectAtIndex:idx];
-    }
+- (void)removeObject:(NSObject*)object {
+    [self removeObject:object inRange:NSRange{ 0, self.count }];
 }
 
 /**
  @Status Interoperable
 */
-- (void)removeObject:(NSObject*)objAddr inRange:(NSRange)range {
-    for (int i = range.location + range.length - 1; i >= (int)range.location; i--) {
-        id curObj = [self objectAtIndex:i];
-
-        if ([curObj isEqual:objAddr]) {
-            [self removeObject:curObj];
+- (void)removeObject:(NSObject*)object inRange:(NSRange)range {
+    NS_COLLECTION_VALIDATE_RANGE(range, self.count);
+    // NSIndexSet can coalesce adjacent ranges. Combined with an efficient removeObjectsInRange:
```

Comment:
removeObjectsInRange doesn't seem especially optimized. Would it be better to just remove them directly here rather than making an IndexSet, iterating and adding indexes to said index set, then iterating over the index set to remove the objects?